test/tests/parallel/set3/mta/account_management.bats

```diff
@@ -0,0 +1,161 @@
+  assert_line --index 2 'user3@localhost.localdomain'
+  assert_line --index 3 'added@localhost.localdomain'
+  assert_line --index 4 'pass@localhost.localdomain'
+  assert_line --index 5 'alias1@localhost.localdomain'
+  # TODO: Probably not intentional?:
+  assert_line --index 6 '@localdomain2.com'
+  _should_output_number_of_lines 7
+
+  # Relevant log output from scripts/helpers/accounts.sh:_create_dovecot_alias_dummy_accounts():
+  # [  DEBUG  ]  Adding alias 'alias1@localhost.localdomain' for user 'user1@localhost.localdomain' to Dovecot's userdb
+  # [  DEBUG  ]  Alias 'alias2@localhost.localdomain' is non-local (or mapped to a non-existing account) and will not be added to Dovecot's userdb
+  # [  DEBUG  ]  Adding alias '@localdomain2.com' for user 'user1@localhost.localdomain' to Dovecot's userdb
+}
+
+@test "should have created maildir for 'user1@localhost.localdomain'" {
+  _run_in_container ls -d '/var/mail/localhost.localdomain/user1'
+  assert_success
+}
+
+@test "should have created maildir for 'user2@otherdomain.tld'" {
+  _run_in_container ls -d '/var/mail/otherdomain.tld/user2'
+  assert_success
+}
+
+@test "should have created maildir for 'user3@localhost.localdomain'" {
+  _run_in_container ls -d '/var/mail/localhost.localdomain/user3'
+  assert_success
+}
+
+@test "should have created maildir for 'added@localhost.localdomain'" {
+  _run_in_container ls -d '/var/mail/localhost.localdomain/added'
+  assert_success
+}
+
+@test "should not accidentally parse comments in 'postfix-accounts.cf' as accounts" {
+  _run_in_container_bash 'ls /var/mail | grep comment'
+  assert_failure
+}
+
+### Account Management ###
+
+@test "should fail to create a user when the domain-part ('@example.com') is missing" {
+  _run_in_container setup email add user_without_domain mypassword
+  assert_failure
+  assert_output --partial 'should include the domain (eg: user@example.com)'
+}
+
+@test "should add new user 'user3@domain.tld' into 'postfix-accounts.cf'" {
+  __should_add_new_user 'user3@domain.tld'
+}
+
+# To catch mistakes from substring matching:
+@test "should add new user 'auser3@domain.tld' into 'postfix-accounts.cf'" {
+  __should_add_new_user 'auser3@domain.tld'
+}
+
+# To catch mistakes from accidental pattern `.` matching as `u`:
+@test "should add new user 'a.ser3@domain.tld' into 'postfix-accounts.cf'" {
+  __should_add_new_user 'a.ser3@domain.tld'
+}
+
+@test "should remove user3 (but not auser3) from 'postfix-accounts.cf'" {
+  # Waits until change event has created directory but not completed:
+  _wait_until_account_maildir_exists 'user3@domain.tld'
+  # Should trigger a new change event:
+  _exec_in_container setup email del -y 'user3@domain.tld'
+
+  # NOTE: This is only checking `postfix-accounts.cf`, account may still persist
+  # elsewhere momentarily such as the Dovecot UserDB until change event kicks in.
+  __check_mail_account_exists 'user3@domain.tld'
+  assert_failure
+  assert_output ''
```

Comment:
The command fails, and we check for empty output. Is that intended?

Reply:
I mostly ported over whatever was in the original test cases. This used to be something like a `[[ -z ${output} ]]` I think. I haven't looked into if it was meant to ensure there wasn't a failure with output.

I can remove the check if it doesn't make sense?

I do recall that if the sed + grep commands didn't handle preventing the substring or `.` wildcard matches, you'd get the wrong result in either of these `__check_mail_account_exists()` calls with output. Although with `assert_failure` I guess this 2nd assert for output doesn't contribute anything? :thinking: 

```suggestion
  assert_failure
```